toolkit/docs/building/add-package.md

```diff
@@ -0,0 +1,436 @@
+            exit 1
+        fi
+        ;;
+        --pkgVersion)
+        if [ -n "$2" ] && [ ${2:0:1} != "-" ]; then
+            PKG_VERSION=$2
+            shift 2
+        else
+            echo "Error: Argument for $1 is missing" >&2
+            exit 1
+        fi
+        ;;
+        --vendorVersion)
+        if [ -n "$2" ] && [ ${2:0:1} != "-" ]; then
+            VENDOR_VERSION=$2
+            shift 2
+        else
+            echo "Error: Argument for $1 is missing" >&2
+            exit 1
+        fi
+        ;;
+        -*|--*=) # unsupported flags
+        echo "Error: Unsupported flag $1" >&2
+        exit 1
+        ;;
+        *) # preserve positional arguments
+        PARAMS="$PARAMS $1"
+        shift
+        ;;
+  esac
+done
+
+echo "--srcTarball      -> $SRC_TARBALL"
+echo "--outFolder       -> $OUT_FOLDER"
+echo "--pkgVersion      -> $PKG_VERSION"
+echo "--vendorVersion   -> $VENDOR_VERSION"
+
+if [ -z "$PKG_VERSION" ]; then
+    echo "--pkgVersion parameter cannot be empty"
+    exit 1
+fi
+
+echo "-- create temp folder"
+tmpdir=$(mktemp -d)
+function cleanup {
+    echo "+++ cleanup -> remove $tmpdir"
+    rm -rf $tmpdir
+}
+trap cleanup EXIT
+
+TARBALL_FOLDER="$tmpdir/tarballFolder"
+mkdir -p $TARBALL_FOLDER
+cp $SRC_TARBALL $tmpdir
+
+pushd $tmpdir > /dev/null
+
+PKG_NAME="ig"
+NAME_VER="$PKG_NAME-$PKG_VERSION"
+VENDOR_TARBALL="$OUT_FOLDER/$NAME_VER-govendor-v$VENDOR_VERSION.tar.gz"
+
+echo "Unpacking source tarball..."
+tar -xf $SRC_TARBALL
+
+echo "Vendor go modules..."
+cd inspektor-gadget-"$PKG_VERSION"
+go mod vendor
+
+echo ""
+echo "========================="
+echo "Tar vendored tarball"
+tar  --sort=name \
+     --mtime="2021-04-26 00:00Z" \
+     --owner=0 --group=0 --numeric-owner \
+     --pax-option=exthdr.name=%d/PaxHeaders/%f,delete=atime,delete=ctime \
+     -cf "$VENDOR_TARBALL" vendor
+
+popd > /dev/null
+echo "$PKG_NAME vendored modules are available at $VENDOR_TARBALL"
+```
+
+We now need to call the script:
+
+```bash
+$ bash generate_source_tarball.sh --srcTarball ig-0.25.0.tar.gz
+...
+$ ls
+generate_source_tarball.sh  ig-0.25.0-govendor-v1.tar.gz  ig.spec  ig-0.25.0.tar.gz
+```
+
+We can now write our signature file:
+
+```bash
+$ sha256sum *.tar.gz
+ba1c9153ee67cd9d1ebd31021ca3848c1099979c229b26bea6ad05470ac486ac  ig-0.25.0-govendor-v1.tar.gz
+3d283e0f6e758b36e291e483e79e7a08c4ce2965a6c11f5c1ab671ced7015a6e  v0.25.0.tar.gz
+# Edit the ig.signatures.json.
+$ $EDITOR ig.signatures.json
+...
+# The content should be the following:
+$	cat ig.signatures.json
+{
+ "Signatures": {
+  "ig-0.25.0-govendor-v1.tar.gz": "ba1c9153ee67cd9d1ebd31021ca3848c1099979c229b26bea6ad05470ac486ac",
+  "ig-0.25.0.tar.gz": "3d283e0f6e758b36e291e483e79e7a08c4ce2965a6c11f5c1ab671ced7015a6e"
+ }
+}
+```
+
+## Building the package
+
+We have all the files needed to build the package, let's build it:
+
+```bash
+$ cd toolkit
+# First, we need to prepare to get the toolchain:
+$ ./scripts/setuplkgtoolchain.sh
```

Comment:
This script is currently only available in the 3.0 tree. Let's target this PR towards 3.0-dev. Eventually, the need for this script will go away when we publish our 3.0 packages to our package server, similar to 2.0, where the toolkit will natively pull packages from our public package repo.